staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -169,16 +175,39 @@ func (s *SecureServingOptions) ApplyTo(c *server.Config) error {
 		c.SecureServingInfo.SNICerts[server.LoopbackClientServerNameOverride] = &tlsCert
 	}
 
-	// create shared informers
-	clientset, err := kubernetes.NewForConfig(c.LoopbackClientConfig)
+	// create shared informers, if not explicitly set use in cluster config.
+	// do not fail on an error, this allows an external API server to startup
+	// outside of a kube cluster.
+	var clientCfg *rest.Config
+	err = nil
+	if s.useLoopbackCfg {
+		clientCfg = c.LoopbackClientConfig
+	} else {
+		clientCfg, err = rest.InClusterConfig()
+	}
 	if err != nil {
-		return err
+		glog.Errorf("Couldn't create in cluster config due to %v. SharedInformerFactory will not be set.", err)
+		return nil
 	}
-	c.SharedInformerFactory = informers.NewSharedInformerFactory(clientset, c.LoopbackClientConfig.Timeout)
-
+	clientset, err := kubernetes.NewForConfig(clientCfg)
+	if err != nil {
+		glog.Errorf("Couldn't create clientset due to %v. SharedInformerFactory will not be set.", err)
+		return nil
+	}
+	c.SharedInformerFactory = informers.NewSharedInformerFactory(clientset, clientCfg.Timeout)
```

Comment:
This is a default resync time, not a normal timeout.  Because of the way shared informers work, its actually a minimum resync time, so I think 10 minutes is what it is in kube at the moment.